client/checkout/classic/new-upe.js

```diff
@@ -0,0 +1,242 @@
+import apiRequest from '../utils/request';
+import {
+	PAYMENT_METHOD_NAME_BANCONTACT,
+	PAYMENT_METHOD_NAME_BECS,
+	PAYMENT_METHOD_NAME_CARD,
+	PAYMENT_METHOD_NAME_EPS,
+	PAYMENT_METHOD_NAME_GIROPAY,
+	PAYMENT_METHOD_NAME_IDEAL,
+	PAYMENT_METHOD_NAME_P24,
+	PAYMENT_METHOD_NAME_SEPA,
+	PAYMENT_METHOD_NAME_SOFORT,
+} from '../constants.js';
+import {
+	getHiddenBillingFields,
+	getSelectedUPEGatewayPaymentMethod,
+	getTerms,
+} from '../utils/upe';
+import showErrorCheckout from '../utils/show-error-checkout';
+import { getAppearance } from '../upe-styles';
+import { getFingerprint } from '../utils/fingerprint';
+
+jQuery( function ( $ ) {
+	const paymentMethods = getUPEConfig( 'paymentMethodsConfig' );
+	const gatewayUPEComponents = {};
+	const api = new WCPayAPI(
+		{
+			publishableKey: getUPEConfig( 'publishableKey' ),
+			accountId: getUPEConfig( 'accountId' ),
+			forceNetworkSavedCards: getUPEConfig( 'forceNetworkSavedCards' ),
+			locale: getUPEConfig( 'locale' ),
+		},
+		apiRequest
+	);
+
+	let fingerprint = null;
+	let appearance = getUPEConfig( 'upeAppearance' );
+
+	enqueueFraudScripts( getUPEConfig( 'fraudServices' ) );
+	initAppearance();
+	initStripeElements();
+
+	function initAppearance() {
+		if ( ! appearance ) {
+			appearance = getAppearance();
+			api.saveUPEAppearance( appearance );
+		}
+	}
+
+	function initStripeElements() {
+		for ( const paymentMethodType in paymentMethods ) {
+			gatewayUPEComponents[ paymentMethodType ] = {
+				elements: null,
+				upeElement: null,
+			};
+		}
+	}
+
+	/**
+	 * This is the entry-point of this file, which is responsible for mounting the UPE elements.
+	 * It is called when the page is loaded.
+	 *
+	 */
+	$( document.body ).on( 'updated_checkout', () => {
+		$( '.wcpay-upe-element' )
+			.toArray()
+			.forEach( ( domElement ) => {
+				const paymentMethodType = $( domElement ).data(
+					'payment-method-type'
+				);
+				const upeElement =
+					gatewayUPEComponents[ paymentMethodType ].upeElement;
+				if ( upeElement ) {
+					upeElement.mount( domElement );
+				} else {
+					createAndMountPaymentElement(
+						paymentMethodType,
+						domElement
+					);
+				}
+			} );
+	} );
+
+	/**
+	 * Block UI to indicate processing and avoid duplicate submission.
+	 *
+	 * @param {Object} $form The jQuery object for the form.
+	 */
+	const blockUI = ( $form ) => {
+		$form.addClass( 'processing' ).block( {
+			message: null,
+			overlayCSS: {
+				background: '#fff',
+				opacity: 0.6,
+			},
+		} );
+	};
+
+	/**
+	 * Creates the UPE element and mounts it to the DOM element.
+	 * No Stripe PaymentIntent is created at this point.
+	 *
+	 * @param {string} paymentMethodType The type of payment method, e.g. 'card', 'giropay', etc.
+	 * @param {string} domElement The selector of the DOM element of particular payment method to mount the UPE element to.
+	 */
+	async function createAndMountPaymentElement(
+		paymentMethodType,
+		domElement
+	) {
+		const options = {
+			mode: 'payment',
```

Comment:
Nice catch - the checkout page failed to load for trial subscriptions since creating elements instance for zero-amount in `payment` mode was declined by Stripe. https://github.com/Automattic/woocommerce-payments/pull/5869/commits/e58a0bbc6a83bd886be40a6a26e829cb0729affe fixes this.